zulipterminal/ui_tools/utils.py

```diff
@@ -22,9 +22,11 @@ def create_msg_box_list(model: Any, messages: Union[None, Iterable[Any]]=None,
     muted_msgs = 0  # No of messages that are muted.
     for msg in message_list:
         # Remove messages of muted topics / streams.
+        # This is not displayed on All messages.
         if is_muted(msg, model):
             muted_msgs += 1
-            continue
+            if not model.narrow:
```

Comment:
I know that this is pythonic for an empty list and the code is not consistent here, but `[]` is an implementation detail of 'all messages', so I think it would be clearer as `if model.narrow == []` ?

Possibly move the added comment to this line too? ie. something like "Don't show in 'All messages'" ?

Reply:
Done. The aim was to make it more 'Pythonic', however as you said this does make code readability a tad tougher.